### PR TITLE
Implement prompt_if_dirty action

### PR DESCRIPTION
**Pull request by jadefish**
Add `prompt_if_dirty` action.

```ruby
content id: :some_content do
  text_field name: :texty_text

  button :save # ...
  button :cancel do
    event :click do
      prompt_if_dirty :some_dialog
      # checkpoint
      replaces :some_content # ...
      hides :some_thing
    end
  end
end
```

In the above sample, `:some_dialog` is shown iff the parent container has a component that is:
* capable of reporting a dirty state, and
* currently reporting a dirty state.

A component is 'dirtyable' if its JS implementation exposes an `isDirty()` method. The component is dirty if the `isDirty` method returns a truthy value.

If none of the container's dirtyable fields are currently dirty, the dialog is not invoked, and the remaining actions are executed as normal.

If a dirtyable field exists and is dirty, a dialog result containing `close`, `halt`, `stay`, `stop`, or `cancel` will cause the dialog to be dismissed and the block to immediately exit without executing the remaining actions.

Otherwise, the remaining actions are executed as normal.

### CODE REVIEW

**Comment by jadefish** [comment]
This isn't quite ready to be merged yet. The `prompt_if_dirty` action should accept an `input_tag` parameter for when a parent container isn't available, but I haven't implemented this functionality yet.

File: views/mdc/assets/js/components/events.js [hunk 1]
```diff
@@ -50,13 +51,19 @@ export class VEvents {
...
                 window.location = responseURL;
             }
-
         }).catch(function(results) {
             console.log('If you got here it may not be what you think:',
                 results);
 
-            var result = results.pop();
-            new VErrors(event).displayErrors(result);
+            let result = results;
```
Comment (nick-geotix) [inline views/mdc/assets/js/components/events.js]: @rx: would like you hear your feedback on this change and the concept of squelching ignorable errors.

File: views/mdc/assets/js/components/events.js [hunk 2]
```diff
@@ -50,13 +51,19 @@ export class VEvents {
...
+
+            if (typeof results.pop === 'function') {
+                result = results.pop();
+            }
+
+            if (!result.squelch) {
```
Comment (jadefish) [inline views/mdc/assets/js/components/events.js]: @rx: would like you hear your feedback on this change and the concept of squelching ignorable errors.

`prompt_if_dirty` rejects the action promise chain's current promise if the user cancels the dialog, which results in this `catch` function being executed despite no real error having occurred.

Comment (rx) [reply views/mdc/assets/js/components/events.js]: As discussed, this seems like useful and good behavior. The ability to add an action that stops the action chain, but does not display an error.

**Pull request merged**